### PR TITLE
Unpin `huggingface_hub`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

- As the release `0.16` is out today.

- Also, use `--upgrade-strategy eager` in `pip install` which is required to respect [this comment](https://github.com/huggingface/transformers/pull/24424#pullrequestreview-1493647494).

  The default `-U` (which is associated with `only-if-needed`) won't upgrade to all available new versions. See [the doc](https://pip.pypa.io/en/stable/development/architecture/upgrade-options/#controlling-what-gets-installed):

  > packages are only upgraded if they are named in the pip command or a requirement file (i.e, they are direct requirements), or an upgraded parent needs a later version of the dependency than is currently installed.

- Since some packages are upgraded, let's change the cache version number for the new versions could be included in the cache.

### CODE REVIEW

File: .circleci/create_circleci_config.py [hunk 1]
```diff
@@ -116,8 +116,6 @@ def to_dict(self):
             },
         ]
         steps.extend([{"run": l} for l in self.install_steps])
-        # TODO (ydshieh): Remove this line after the next release (the one after 2023/06/19) of `huggingface_hub`
```
Comment (ydshieh) [inline .circleci/create_circleci_config.py:119]: This is the main objective of this PR.

File: .circleci/config.yml [hunk 2]
```diff
@@ -30,9 +30,9 @@ jobs:
         parallelism: 1
         steps:
             - checkout
-            - run: pip install --upgrade pip
-            - run: pip install -U GitPython
-            - run: pip install -U .
+            - run: pip install --upgrade --upgrade-strategy eager pip
```
Comment (ydshieh) [inline .circleci/config.yml:33]: so we can get newer versions when they are available.

File: .circleci/config.yml [hunk 3]
```diff
@@ -136,20 +136,20 @@ jobs:
             - checkout
             - restore_cache:
                   keys:
-                      - v0.6-code_quality-{{ checksum "setup.py" }}
-                      - v0.6-code-quality
+                      - v0.7-code_quality-{{ checksum "setup.py" }}
```
Comment (ydshieh) [inline .circleci/config.yml:139]: to include the new packages' versions (installed by this PR)

**Review by amyeroberts — APPROVED** [review]
Thanks for updating!

**Comment by Wauplin** [comment]
Thank you @ydshieh !

**Pull request merged**